core/web/resolver/query.go

```diff
@@ -68,7 +69,7 @@ func (r *Resolver) Chain(ctx context.Context, args struct{ ID graphql.ID }) (*Ch
 		return nil, err
 	}
 
-	cs, _, err := r.App.EVMORM().Chains(0, -1, string(args.ID))
+	cs, _, err := r.App.EVMORM().Chains(relay.ChainID(args.ID))
```

Comment:
Did you consider using an alias for `ChainID`?
```go
-type ChainID string
+type ChainID = string
```
Then you could drop all these casts, but would have to give up the method, so maybe that defeats the purpose of the custom type?

Reply:
Maybe there is an opportunity to use a generic constraint like `~string`?

Reply:
i considered the other way around -- declaring the config struct to have an `id` of type relay.ChainID and sorting out the TOML de/serialization, but that seemed a step too far for this change.

i don't think that generics are a good fit here because there are so few call sites.

the motivation for this change is pre-saging https://smartcontract-it.atlassian.net/browse/BCF-2507

but admittedly it doesn't go all the way to fixing it. 

i think the public APIs should take a type, as described in that ticket. For this change i could
1. do nothing -- revert to using strings in the API and leave it for another change
2. leave the changes in place
3. modify the internal helper signatures to take a string inside of a ChainID. this feels weird to me but would reduce the number of casts.

Reply:
> i don't think that generics are a good fit here because there are so few call sites.

But the tradeoff is dropping many needless conversions and method calls.

Why does `type ChainID` even exist? If it is only to support the oddball EVM case, then why not just let EVM code handle that exception, so the other families are unaffected? If it is meant to support other chain family formats in the future, then adding method for each one is not going to scale - why doesn't `~string` fit in that case?

Reply:
changed to be an alias and mv'd the validation into evm specific code